File: padroesCriacionais/padrao-de-projeto-prototype/java/src/main/java/Map.java
public class Map implements Cloneable {
    public String name;
    public String minimumLevel;
    public String coordinateNpcX;
    public String coordinateNpcY;


    public Map(
            String name,
            String minimumLevel,
            String coordinateNpcX,
            String coordinateNpcY
    ) {
        this.name = name;
        this.minimumLevel = minimumLevel;
        this.coordinateNpcX = coordinateNpcX;
        this.coordinateNpcY = coordinateNpcY;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMinimumLevel() {
        return minimumLevel;
    }

    public void setMinimumLevel(String minimumLevel) {
        this.minimumLevel = minimumLevel;
    }

    public String getCoordinateNpcX() {
        return this.coordinateNpcX;
    }

    public void setCoordinateNpcX(String coordinateX) {
        this.coordinateNpcX = coordinateX;
    }

    public String getCoordinateNpcY() {
        return this.coordinateNpcY;
    }
    public void setCoordinateNpcY(String coordinateY) {
        this.coordinateNpcY = coordinateY;
    }

    @Override
    public Map clone() throws CloneNotSupportedException {
        Map mapClone = (Map) super.clone();
        return mapClone;
    }

    @Override
    public String toString() {
        return "Map{" +
                "name='" + name +
                "', level minimo='" + minimumLevel +
                "', coordenada x='" + this.coordinateNpcX + '\'' +
                ", coordenada y='" + this.coordinateNpcY + '\'' +
                '}';
    }
}

File: padroesCriacionais/padrao-de-projeto-prototype/php/src/Map.php
<?php

namespace Prototype\Php;

class Map
{
    public function __construct(
        private String $name,
        private String $minimumLevel,
        private String $coordinateNpcX,
        private String $coordinateNpcY
    ) {}

    /**
     * @return String
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param String $name
     */
    public function setName(string $name): void
    {
        $this->name = $name;
    }

    /**
     * @return String
     */
    public function getMinimumLevel(): string
    {
        return $this->minimumLevel;
    }

    /**
     * @param String $minimumLevel
     */
    public function setMinimumLevel(string $minimumLevel): void
    {
        $this->minimumLevel = $minimumLevel;
    }

    /**
     * @return String
     */
    public function getCoordinateNpcX(): string
    {
        return $this->coordinateNpcX;
    }

    /**
     * @param String $coordinateNpcX
     */
    public function setCoordinateNpcX(string $coordinateNpcX): void
    {
        $this->coordinateNpcX = $coordinateNpcX;
    }

    /**
     * @return String
     */
    public function getCoordinateNpcY(): string
    {
        return $this->coordinateNpcY;
    }

    /**
     * @param String $coordinateNpcY
     */
    public function setCoordinateNpcY(string $coordinateNpcY): void
    {
        $this->coordinateNpcY = $coordinateNpcY;
    }

    public function __toString(): string
    {
        return "Map{" .
            "name='" . $this->name .
            "', level minimo='" . $this->minimumLevel .
            "', coordenada x='" . $this->coordinateNpcX . '\'' .
            ", coordenada y='" . $this->coordinateNpcY . '\'' .
            '}';
    }
}